subsys/net/lib/http/http_hpack.c

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (c) 2023, Emna Rekik
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <errno.h>
+#include <string.h>
+
+#include <zephyr/net/http/hpack.h>
+
+static inline bool http_hpack_key_is_static(int32_t key)
+{
+	return key > HTTP_SERVER_HPACK_INVALID && key <= HTTP_SERVER_HPACK_SIZE;
+}
+
+static inline bool http_hpack_key_is_dynamic(int32_t key)
+{
+	return key > HTTP_SERVER_HPACK_SIZE;
```

Comment:
`HTTP_SERVER_HPACK_SIZE` here is actually incorrect, and I believe Robert pointed that out over a week ago. I believe the check should use `HTTP_SERVER_HPACK_WWW_AUTHENTICATE`.

Please use `uint32_t` instead of `int32_t`.